src/app/settings/sponsored-families.component.ts

```diff
@@ -1,13 +1,33 @@
 import { Component, OnInit } from "@angular/core";
+import {
+  AsyncValidatorFn,
+  FormBuilder,
+  FormGroup,
+  ValidationErrors,
+  Validators,
+} from "@angular/forms";
 
 import { ApiService } from "jslib-common/abstractions/api.service";
 import { I18nService } from "jslib-common/abstractions/i18n.service";
 import { OrganizationService } from "jslib-common/abstractions/organization.service";
 import { PlatformUtilsService } from "jslib-common/abstractions/platformUtils.service";
+import { StateService } from "jslib-common/abstractions/state.service";
 import { SyncService } from "jslib-common/abstractions/sync.service";
 import { PlanSponsorshipType } from "jslib-common/enums/planSponsorshipType";
 import { Organization } from "jslib-common/models/domain/organization";
 
+function notCurrentUserEmailValidator(stateService: StateService): AsyncValidatorFn {
+  return async (control): Promise<ValidationErrors | null> => {
+    const currentUserEmail = await stateService.getEmail();
+
+    if (control.value.toLowerCase() === currentUserEmail.toLowerCase()) {
+      return {
+        cannotSponsorSelf: true,
+      };
+    }
+  };
+}
+
```

Comment:
Would the string parameter be a replacement for `await stateService.getEmail();` or `cannotSponsorSelf`? I think it we "generify" it we would want a `ValidationErrors` parameter for the error the user wants and an async function so the caller can define the call they want.

Reply:
I was thinking it would be a `ValueNotAllowedValidator` and the string parameter would either be regex pattern or a specific string that isn't allowed, then we would just apply the `valueNotAllowed` error to the control rather than one this specific.

Reply:
Moved to a more generic validator.